torch/csrc/jit/codegen/cuda/lower_expr_sort.cpp

```diff
@@ -942,17 +986,56 @@ bool ExprSegmentationSorter::supportedMerge(ExprGroup* sg1, ExprGroup* sg2) {
     return false;
   }
 
-  auto producer_domain = producer_group->payload()->ca_domains_;
-  auto consumer_domain = consumer_group->payload()->pa_domains_;
+  auto producer_ca_domain = producer_group->payload()->ca_domains_;
```

Comment:
Can you please add a comment describing the basic logic behind `supportedMerge`?